### PR TITLE
DOC: Fix indentation in docstring example

### LINKED ISSUE #30830: DOC: Fix indentation in docstring example
Not sure why we didn't see it before, but the example in the first code block of the docstring guidelines seem to be missing the indentation: https://pandas.io/docs/development/contributing_docstring.html#about-docstrings-and-standards

We should fix it, and indent the function properly: https://github.com/pandas-dev/pandas/blame/master/doc/source/development/contributing_docstring.rst#L25

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [x] closes #30830
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: doc/source/development/contributing_docstring.rst [hunk 1]
```diff
@@ -22,38 +22,38 @@ Next example gives an idea on how a docstring looks like:
...
+        Examples
+        --------
+        >>> add(2, 2)
+        4
+        >>> add(25, 0)
+        25
+        >>> add(10, -10)
+        0
+        """
     return num1 + num2
```
Comment (jbrockmendel) [inline doc/source/development/contributing_docstring.rst]: does this line need an indent too?

Comment (jorisvandenbossche) [inline doc/source/development/contributing_docstring.rst]:
```suggestion
        return num1 + num2
```

**Comment by MomIsBestFriend** [comment]
Thank you :) 
@jorisvandenbossche

File: doc/source/development/contributing_docstring.rst (hunk 1 above)
Comment (MomIsBestFriend) [reply doc/source/development/contributing_docstring.rst]: Yes it does, somehow missed it.

**Review by TomAugspurger — APPROVED** [review]
Thanks!

**Pull request merged**